### PR TITLE
Edits to ForceSwitchOutAttr to apply contact effects before switching

### PR DESCRIPTION
**Pull request by EmoUsedHM01**
Should fix issue #83

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -2464,94 +2464,118 @@ export class AddArenaTrapTagAttr extends AddArenaTagAttr {
 }
 
 export class ForceSwitchOutAttr extends MoveEffectAttr {
-  private user: boolean;
-  private batonPass: boolean;
+    private user: boolean;
+    private batonPass: boolean;
```
Comment (molker) [inline src/data/move.ts:2468]: remove indents

**Review by molker — CHANGES_REQUESTED** [review]
the whole thing has been indented one extra tab, please restore the formatting

File: src/data/move.ts [hunk 2]
```diff
@@ -2464,94 +2464,118 @@ export class AddArenaTrapTagAttr extends AddArenaTagAttr {
...
+            resolve(true);
+        });
     }
-    return ret;
-  }
+
+	// Function to apply any effects to the user/target before switching out
+	applyEffectsBeforeSwitchOut(user: Pokemon, target: Pokemon, move: Move) {
+		// Create an instance of PostDefendContactApplyStatusEffectAbAttr
+		const postDefendAttr = new PostDefendContactApplyStatusEffectAbAttr(chance, ...effects);
```
Comment (molker) [inline src/data/move.ts:2541]: need to import PostDefendContactApllyStatusEffectAbAttr

Comment (molker) [reply src/data/move.ts:2541]: also where are you getting chance and ...effects

File: src/data/move.ts [hunk 3]
```diff
@@ -2464,94 +2464,118 @@ export class AddArenaTrapTagAttr extends AddArenaTagAttr {
...
+	
+		// Apply the effects based on the logic in PostDefendContactApplyStatusEffectAbAttr
+		postDefendAttr.applyPostDefend(user, target, move, hitResult, args);
```
Comment (molker) [inline src/data/move.ts:2544]: where are hitResult and args coming from?

File: src/data/move.ts (hunk 1 above)
Comment (EmoUsedHM01) [reply src/data/move.ts:2468]: Are the indents a personal preference thing, or just something that plays badly with typescript?

**Pull request closed without merging**